src/workerd/api/js-rpc-target.c++

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include <workerd/api/js-rpc-target.h>
+#include <workerd/api/global-scope.h>
+
+namespace workerd::api {
+
+kj::Promise<void> JsRpcTargetImpl::call(CallContext callContext) {
+  auto methodName = kj::heapString(callContext.getParams().getMethodName());
+  // TODO(now): Deserialize the arguments.
+  //    - Current demo doesn't take args, once we pass args on the client side I'll deserialize
+  //      on the server side.
+  auto serializedArgs = callContext.getParams().getSerializedArgs().getV8Serialized().asBytes();
+
+  // Try to execute the requested method.
+  try {
+    auto& ctx = incomingRequest->getContext();
+    co_await ctx.run(
+        [methodName=kj::mv(methodName),
+         serializedArgs = kj::mv(serializedArgs),
+         entrypointName = kj::mv(entrypointName),
```

Comment:
By moving away `entryPoint` name here, you're setting the member variable of `JsRpcTargetImpl` to null, so a second call won't work.

Reply:
FWIW you don't need `kj::mv` here -- `Maybe<StringPtr>` can just be copied.

Reply:
Oh right. Yes you're right, as discussed previously we actually do want each call on the top-level stub to be a separate event / session. So my comment here is irrelevant (but it's still true that `kj::mv()` is not needed for `Maybe<StringPtr>`).

Reply:
So I'd like to put forward my view on the session model here, I suspect I disagree about a few things but we also haven't necessarily been clear about terms. I think we have different pieces associated with a js rpc event:
- There is a capnp target interface. I don't see why we should throw this away once we make it, we might as well persist it along with the client stub. It even feels somewhat like we should be extending and existing part of our capability chain with the target interface since we're using the "root". That said, it doesn't really matter all that much from a behavioral perspective. (I do wonder about perhaps providing an abort controller that notices if the client disconnects?)
- There is the metrics reporting via `RequestObserver` and `ActorObserver`. I believe that our recording is much more efficient via the `ActorObserver` for web sockets than it is through the `RequestObserver` for regular fetches both in terms of total events and in terms of relevance of fields. I'm concerned that someone will see our "local object" sugar and try to make a multi producer single consumer queue that overwhelms our metrics infrastructure. They could do that with fetch, but there are more expectations on what is reasonable with http shaped rpc.
- There is also the `IncomingRequest` which only really matters here in so far as it is the glue between the other two.

So what's a "session" here? I think I'd like us to think of the client stub as a session. (Hopefully, we agree that each call on a target produces an "event".)

Reply:
Ben and I spoke about this briefly; it looks like we wouldn't avoid the `RequestObserver` by using the Loopback mechanism used for hibernatable websockets since we create a new request for each hibernatable websocket message.

Given this would require more work than simply switching to loopback, we're going to continue with the request/response (no session) model and keep an eye on rate of metrics. We can revisit this for JS RPC (and hibernatable websockets) in the future if need be.

Reply:
https://github.com/cloudflare/workerd/pull/1028/commits/53d4434853c83be086daf9f67560fe590f4c3a08

Reply:
I don't really follow the exchange in this thread since my last comment.

I think it's important that the existence of a stub object pointing at it doesn't prevent the DO from hibernating, even if that stub has been used for one or more RPC calls in the past.